gui/layers/_vectors_layer/model.py

```diff
@@ -0,0 +1,615 @@
+        length.setter adjusts the underlying data
+        subscribe an observer by registering it with "length_bind_to"
+
+    color : str
+        one of "get_color_names" from vispy.color
+
+    conector : str or np.ndarray
+        way to render line between coordinates
+        two built in types: "segment" or "strip"
+        third type is np.ndarray
+
+    mode : str
+        control panel mode
+
+    Attributes
+    ----------
+    Private attributes
+    -------
+    _kernel_dict
+    _kernel
+    _data_types
+    _data_type
+    _width
+    _color
+    _colors
+    _connector_types
+    _connector
+    _avg_dims
+    _averaging
+    _length
+    _avg_observers
+    _len_observers
+    _need_display_update
+    _need_visual_update
+    _raw_dat
+    _original_data
+    _current_shape
+    _vectors
+    _mode
+    _mode_history
+
+    Public attributes
+    -------
+    name
+
+
+    See vispy's line visual docs for more details:
+    http://api.vispy.org/en/latest/visuals.html#vispy.visuals.LineVisual
+    http://vispy.org/visuals.html
+    """
+
+    def __init__(
+        self, vectors,
+            width=1,
+            color='red',
+            connector='segments',
+            averaging='1x1',
+            length=1):
+
+        visual = LinesNode()
+        super().__init__(visual)
+
+        # map averaging type to tuple
+        self._kernel_dict = {'1x1': (1, 1),
+                             '3x3': (3, 3),
+                             '5x5': (5, 5),
+                             '7x7': (7, 7),
+                             '9x9': (9, 9),
+                             '11x11': (11, 11)}
+        self._kernel = self._kernel_dict['1x1']
+
+        # Store underlying data model
+        self._data_types = ('matrix', 'projection')
+        self._data_type = None
+
+        # Save the line style params
+        self._width = width
+        self._color = color
+        self._colors = get_color_names()
+        self._connector_types = ['segments']
+        self._connector = connector
+
+        # averaging and length attributes
+        self._avg_dims = ['1x1', '3x3', '5x5', '7x7', '9x9', '11x11']
+        self._averaging = averaging
+        self._length = length
+        self._avg_observers = []
+        self._len_observers = []
+
+        # update flags
+        self._need_display_update = False
+        self._need_visual_update = False
+
+        # assign vector data and establish default behavior
+        self._raw_dat = None
+        self._original_data = vectors
+        self._current_shape = vectors.shape
+        self._vectors = self._check_vector_type(vectors)
+        self.averaging_bind_to(self._default_avg)
+        self.length_bind_to(self._default_length)
+
+        self._mode = 'pan/zoom'
+        self._mode_history = self._mode
+
+        self.name = 'vectors'
+        self.events.add(mode=Event)
+        self._qt_properties = QtVectorsLayer(self)
+        self._qt_controls = QtVectorsControls(self)
```

Comment:
See my comment around `set_mode` on whether you can delete this line too. `_qt_controls` is set to `None` in the base class, and I just checked that everything will launch fine without it